### PR TITLE
Document ScrollPosition

### PR DESCRIPTION
**Pull request by abarth**

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scroll_position.dart [hunk 1]
```diff
@@ -18,14 +18,45 @@ import 'scroll_metrics.dart';
...
+/// Over time, a [Scrollable] might have many different [ScrollPosition]
+/// objects. For example, if [Scrollable.physics] changes, [Scrollable] creates
+/// a new [ScrollPosition] with the new physics. To transfer state from the old
+/// instance to the new instance, subclasses implement [absorb]. See [absorb]
+/// for more details.
+///
+/// Subclasses also need to call [didUpdateScrollDirection] whenever
+/// [userScrollDirection] changes values.
+///
+/// See also:
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_position.dart]: should the see also be above the subsection header?

Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_position.dart]: (what did we do for RenderObject?)

File: packages/flutter/lib/src/widgets/scroll_position.dart [hunk 2]
```diff
@@ -18,14 +18,45 @@ import 'scroll_metrics.dart';
...
+/// The [pixels] value determines the scroll offset that the scroll view uses to
+/// select which part of its content to display. As the user scrolls the
+/// viewport, this value changes, which changes the content that is displayed.
+///
+/// The [ScrollPosition] applies [physics] to scrolling, and stores the
+/// [minScrollExtent] and [maxScrollExtent].
+///
+/// Scrolling is controlled by the current [activity], which is set by
+/// [beginActivity]. [ScrollPosition] itself does not start any activities.
+/// Instead, concrete subclasses, such as [ScrollPositionWithSingleContext]
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_position.dart]: either remove the comma before `such` or add one at the end of the line

File: packages/flutter/lib/src/widgets/scroll_position.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/scroll_position.dart]: RenderObject has no "See also" section apparently.

File: packages/flutter/lib/src/widgets/scroll_position.dart [hunk 3]
```diff
@@ -18,14 +18,45 @@ import 'scroll_metrics.dart';
...
+/// typically start activities in response to user input or instructions from a
+/// [ScrollController].
+///
+/// This object notifies its listeners when [pixels] changes.
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_position.dart]: this leaves open the question of how; maybe "This object is a Listenable that notifies..."?

File: packages/flutter/lib/src/widgets/scroll_position.dart [hunk 4]
```diff
@@ -18,14 +18,45 @@ import 'scroll_metrics.dart';
...
+///
+/// ## Subclassing ScrollPosition
+///
+/// Over time, a [Scrollable] might have many different [ScrollPosition]
+/// objects. For example, if [Scrollable.physics] changes, [Scrollable] creates
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_position.dart]: changes type

File: packages/flutter/lib/src/widgets/scroll_position.dart [hunk 5]
```diff
@@ -292,21 +378,75 @@ abstract class ScrollPosition extends ViewportOffset with ScrollMetrics {
...
+  /// prematurely (e.g. from another activity taking over). See
+  /// [ScrollDragController.onDragCanceled] for details.
   Drag drag(DragStartDetails details, VoidCallback dragCancelCallback);
 
+  /// The currently operative [ScrollActivity].
+  ///
+  /// If the scroll position is not performing any more specific activity, the
+  /// activity will be an [IdleScrollActivity]. To determine whether the scroll
+  /// position is idle, check whether [userScrollDirection] is
+  /// [ScrollDirection.idle].
```
Comment (Hixie) [inline packages/flutter/lib/src/widgets/scroll_position.dart]: userScrollDirection will be idle when animating.

**Comment by Hixie** [comment]
LGTM

File: packages/flutter/lib/src/widgets/scroll_position.dart (hunk 5 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/scroll_position.dart]: Changed to isScrollingNotifier instead.

File: packages/flutter/lib/src/widgets/scroll_position.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/scroll_position.dart]: Left as-is.

**Pull request merged**